app/models/service_ansible_playbook.rb

```diff
@@ -1,17 +1,74 @@
 class ServiceAnsiblePlaybook < ServiceGeneric
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, add_options = {})
+    save_job_options(action, add_options)
+  end
+
   def execute(action)
-    _log.info("Execute for Service context: #{action}")
+    jt = job_template(action)
+    opts = get_job_options(action)
+
+    _log.info("Launching Ansible Tower job with options: #{opts}")
+    new_job = ManageIQ::Providers::AnsibleTower::AutomationManager::Job.create_job(jt, opts)
+
+    _log.info("Ansible Tower job with ref #{new_job.ems_ref} was created.")
+    add_resource!(new_job, :name => action)
   end
 
   def check_completed(action)
-    _log.info("Check_completed for Service context: #{action}")
+    status, reason = job(action).raw_status.normalized_status
+    done    = status != 'transient'
+    message = status == 'create_complete' ? nil : reason
+    [done, message]
+  rescue MiqException::MiqOrchestrationStackNotExistError, MiqException::MiqOrchestrationStatusError => err
+    [true, err.message] # consider done with an error when exception is caught
   end
 
   def refresh(action)
-    _log.info("Refresh for Service context: #{action}")
+    job(action).refresh_ems
+  end
+
+  def check_refreshed(_action)
+    [true, nil]
+  end
+
+  def job(action)
+    service_resources.find_by!(:name => action, :resource_type => 'OrchestrationStack').try(:resource)
   end
 
-  def check_refreshed(action)
-    _log.info("Check_refreshed for Service context: #{action}")
+  private
+
+  def job_template(action)
+    service_template.resource_actions.find_by!(:action => action).configuration_template
+  end
+
+  def get_job_options(action)
+    options["#{action.downcase}_job_options".to_sym].deep_dup
+  end
+
+  def save_job_options(action, overrides)
+    job_options = parse_dialog_options
+    job_options[:extra_vars] = (job_options[:extra_vars] || {}).merge(overrides[:extra_vars]) if overrides[:extra_vars]
+    job_options.merge!(overrides.except(:extra_vars))
+
+    options["#{action.downcase}_job_options".to_sym] = job_options
+    save!
+  end
+
+  def parse_dialog_options
+    dialog_options = options[:dialog] || {}
+    {
+      :credential => dialog_options['dialog_credential'],
+      :hosts      => dialog_options['dialog_hosts']
+    }.compact.merge(extra_vars_from_dialog)
```

Comment:
As discussed this section will need to be reworked.  Credentials will need to be translated from our db ID to the value we need to pass to Ansible.  Also, if the `hosts` value from the dialog is empty I think we would want to skip setting this key.

Can update handled in a later PR.